### ISSUE TITLE
server: new pages are not rendered if page file is created with content (hugo master)

### ISSUE THREAD

**Issue by TiGR** [gohugoio/hugo#12114]
[Example website](https://github.com/TiGR/hugo-testing/tree/broken-getterms-issue). In order to reproduce the issue, you need to do the following:

1. Create a directory for new page/bundle (hugo runs a build)
2. Copy page from somewhere, or create file by writing its content (do not create empty file first) (hugo runs another build)

It works like this:

```
hugo server
mkdir content/test-page-2/
cp content/{test-page,test-page-2}/index.md
```

At this point you should see the new page in the list. But it is not there. And it won't appear until you restart hugo. Tested on master version.

**Comment by TiGR** [comment]
I see CantReproduce tag added.

I've tried it on current master version, and it still works for me (on Windows 11 and on Ubuntu 22.04.

1. Clone the repository: 
    ```bash
    git clone -b broken-getterms-issue https://github.com/TiGR/hugo-testing.git
    cd hugo-testing
    ```
2. Run the server:
    ```bash
    hugo server
    ```
3. Create new page directory (if you skip this step, it won't reproduce):
    ```bash
    mkdir content/test-page-2/
    ```
4. Create new page:
    ```bash
    cp content/test-page/index.md content/test-page-2/index.md
    ```
5. Check the browser, it should list 2 pages, but it only lists one.
6. Restart the server, check browser and notice 2 pages.

I've followed these steps and was able to reproduce it on 2 separate machines.

Also, I've tried it on 0.122 and it works fine, the page appears as it should.
